### PR TITLE
Haptic Feedback Support.

### PR DESCRIPTION
**Pull request by Gamer64ytb**
When you choose a value in the option, the controls will emit a small vibration when pressed, it is not perfect but it is usable.

### CODE REVIEW

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by TheOG00** [comment]
good

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java [hunk 1]
```diff
@@ -896,6 +903,57 @@ public void onStopTrackingTouch(SeekBar seekBar)
...
+      setHapticFeedbackLevel(defaultHapticLevel);
+    });
+
+    builder.create().show();
+  }
+
+  private void setHapticFeedbackLevel(int hapticFeedbackLevel) {
+    SharedPreferences.Editor editor = mPreferences.edit();
+    editor.putInt("hapticFeedbackLevel", hapticFeedbackLevel);
+    editor.apply();
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java:954]: We're trying to migrate away from using `SharedPreferences`. Please add your new setting here instead: https://github.com/dolphin-emu/dolphin/blob/ffdc8538a162b1ca56e3943c5a78c02358053b2a/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/model/IntSetting.java#L27-L32

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java [hunk 2]
```diff
@@ -297,6 +299,15 @@ public boolean onTouch(View v, MotionEvent event)
...
       }
+
+      // Not all devices that support VibrationEffect (API>=26) allow Amplitude Control
+      if (mPreferences.getInt("hapticFeedbackLevel", 0) >= 1 &&
+        android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+        Vibrator vibratorManager = (Vibrator)
+          getContext().getSystemService(Context.VIBRATOR_SERVICE);
+        vibratorManager.vibrate(VibrationEffect.createOneShot(80,
+          mPreferences.getInt("hapticFeedbackLevel", 1)));
+    }
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java:310]: You should include a fallback for older API levels. Preferably vibrating without amplitude control, or if that isn't appropriate, making the option not show up in the GUI.

**Review by JosJuice — COMMENTED** [review]
Please fix the errors reported by the lint bot.

**Comment by JosJuice** [comment]
@Gamer64ytb Would you like to update this PR? If not, we'll close it eventually, so that we don't keep piling up stale PRs. (Even if it does get closed, you're welcome to open a new version of it later.)

**Comment by jordan-woyak** [comment]
This PR has become stale so I will close it for now. We can re-open it upon request.

**Pull request closed without merging**